apollo-backend/src/main/java/io/logz/apollo/controllers/DeploymentController.java

```diff
@@ -188,6 +194,27 @@ public void fetchPaginatedDeploymentHistory(Boolean descending, int pageNumber,
         }
     }
 
+    @LoggedIn
+    @PUT("/force-cancel")
+    public void cancelExpiredDeployment(String id, Req req) {
+        Deployment deployment = deploymentDao.getDeployment(Integer.parseInt(id));
+        switch (deployment.getStatus()) {
```

Comment:
Other endpoints behave like that too, and this scenario shouldn't happen because the id come from the UI from an existing deployment. What do you suggest?

Reply:
I would add a null check for deployment. If you're sure that we always get a valid id, then it's ok.

Do we have an appropriate functionality in apollo-client?

Reply:
yes, we have cancelDeployment method in ApolloClient